core/client/index.ts

```diff
@@ -74,16 +84,50 @@ export class Core {
    *
    * @private
    */
-  async sendMessage(dstIdkeys: string[], payload: string) {
+  async sendMessage(dstIdkeys: string[], messagePayload: string) {
     let batch: outboundEncPayloadType[] = new Array();
   
+    let ownId = this.olmWrapper.getIdkey();
+    let consistencyLoopback = true;
+    for (let r of dstIdkeys) {
+      if (r === ownId) {
+        consistencyLoopback = false;
+        break;
+      }
+    }
+
+    if (consistencyLoopback) {
+      dstIdkeys.push(ownId);
+    }
+
```

Comment:
Everything from here to line 143, except for encrypting the payload body, should be done in the message chain library, with a single call.